UpdateDependencies.ps1

```diff
@@ -0,0 +1,129 @@
+﻿#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+function EnsureEnvironmentVariable([string]$envVarName)
+{
+    If ([Environment]::GetEnvironmentVariable($envVarName) -eq $null)
+    {
+        throw "Can't find environment variable '$envVarName'"
+    }
+}
+
+function SetEnvIfDefault([string]$envVarName, [string]$value)
+{
+    If ([Environment]::GetEnvironmentVariable($envVarName) -eq $null)
+    {
+        [Environment]::SetEnvironmentVariable($envVarName, $value)
+    }
+}
+
+EnsureEnvironmentVariable "GITHUB_USER"
+EnsureEnvironmentVariable "GITHUB_EMAIL"
+EnsureEnvironmentVariable "GITHUB_PASSWORD"
+
+SetEnvIfDefault 'COREFX_VERSION_URL' 'https://raw.githubusercontent.com/dotnet/versions/master/dotnet/corefx/master/Latest.txt'
+SetEnvIfDefault 'GITHUB_ORIGIN_OWNER' $env:GITHUB_USER
+SetEnvIfDefault 'GITHUB_UPSTREAM_OWNER' 'dotnet'
+SetEnvIfDefault 'GITHUB_PROJECT' 'corefx'
+SetEnvIfDefault 'GITHUB_UPSTREAM_BRANCH' 'master'
+
+
+# Updates the dir.props file with the latest CoreFX build number
+function UpdateValidDependencyVersionsFile
+{
+    $CoreFxLatestVersion = Invoke-WebRequest $env:COREFX_VERSION_URL -UseBasicParsing
+    $CoreFxLatestVersion = $CoreFxLatestVersion.ToString().Trim()
+
+    if (!$CoreFxLatestVersion)
+    {
+        Write-Error "Unable to find latest CoreFX version at $env:COREFX_VERSION_URL"
+        return $false
+    }
+
+    $DirPropsPath = "$PSScriptRoot\dir.props"
+    $xml = new-object xml
+    $xml.PreserveWhitespace = $true
+    $xml.Load($DirPropsPath)
+
+    $nsMgr = new-object Xml.XmlNamespaceManager $xml.NameTable
+    $nsMgr.AddNamespace('b', 'http://schemas.microsoft.com/developer/msbuild/2003')
+
+    $node = $xml.SelectSingleNode('//b:ValidationPattern[@Include = "CoreFxVersions"]/b:ExpectedPrerelease', $nsMgr)
```

Comment:
The dir.props does get slightly reformatted by this, but there are only 2 changes. You can see what it would look like here:  https://github.com/eerhardt/corefx/commit/0693e38152333d6f0f6c8ba12b029f994a93a363#diff-0b192804a6349e8c26d2b027afbd89a2L53

There are only two minor changes:

```DIFF
-  <Import Project="$(MSBuildThisFileDirectory)Packaging.props"/>
+  <Import Project="$(MSBuildThisFileDirectory)Packaging.props" />
```

```DIFF
-    <DnuRestoreSource>@(DnuSourceList -> '--source %(Identity)', ' ')</DnuRestoreSource>
+    <DnuRestoreSource>@(DnuSourceList -&gt; '--source %(Identity)', ' ')</DnuRestoreSource>
```

My feeling was to not add a level of indirection (i.e. pull the value out into its own property) just for this script. Instead, write the dir.props as it makes sense, and then make the automation conform to it.

I can totally change this to to pull it out into its own variable, if you are strongly against this approach.